### PR TITLE
Sonar fixes carbon identity framework

### PR DESCRIPTION
**Pull request by supunharsha**

### CODE REVIEW

File: components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/KeyStoreAdmin.java [hunk 1]
```diff
@@ -292,7 +292,12 @@ public void addTrustStore(byte[] content, String filename, String password, Stri
...
-            String keyStoreNameTrim = keyStoreName.trim();
+
+            String keyStoreNameTrim = null;
+            if (keyStoreName != null) {
+                keyStoreNameTrim = keyStoreName.trim();
+            }
+
             if (keyStoreName == null || keyStoreNameTrim.length() == 0) {
                 throw new SecurityConfigException("Key Store name can't be null");
             }
```
Comment (supunharsha) [inline components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/KeyStoreAdmin.java]: put keyStoreName here

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java [hunk 2]
```diff
@@ -522,7 +518,7 @@ private void mapSPClaimsAndFilterRequestedClaims(Map<String, String> spToLocalCl
             } catch (Exception e) {
                 throw new FrameworkException("Error occurred while getting all claim mappings from " +
                         ApplicationConstants.LOCAL_IDP_DEFAULT_CLAIM_DIALECT + " dialect to " +
-                         spStandardDialect+ " dialect for " +
+                         spStandardDialect + " dialect for " +
```
Comment (supunharsha) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java]: Add two line together

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java [hunk 3]
```diff
@@ -549,22 +545,26 @@ private void mapSPClaimsAndFilterRequestedClaims(Map<String, String> spToLocalCl
...
+                        entry.setValue(roleClaim);
                     }
-                    String roleClaim = entry.getValue();
-                    List<String> rolesList = new LinkedList<>(Arrays.asList(roleClaim.split(claimSeparator)));
-                    roleClaim = getServiceProviderMappedUserRoles(appConfig, rolesList, claimSeparator);
-                    entry.setValue(roleClaim);
                 }
             }
 
+
```
Comment (supunharsha) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java]: unnecessary new lines

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java [hunk 4]
```diff
@@ -549,22 +545,26 @@ private void mapSPClaimsAndFilterRequestedClaims(Map<String, String> spToLocalCl
...
 
-            for (Map.Entry<String, String> entry : allLocalClaims.entrySet()) {
-                //set local2sp role mappings
-                if (FrameworkConstants.LOCAL_ROLE_CLAIM_URI.equals(entry.getKey())) {
-                    RealmConfiguration realmConfiguration = userStore.getRealmConfiguration();
-                    String claimSeparator = realmConfiguration
-                            .getUserStoreProperty(IdentityCoreConstants.MULTI_ATTRIBUTE_SEPARATOR);
-                    if (StringUtils.isBlank(claimSeparator)) {
-                        claimSeparator = IdentityCoreConstants.MULTI_ATTRIBUTE_SEPARATOR_DEFAULT;
+            if (allLocalClaims != null) {
```
Comment (supunharsha) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java]: If null return new hash map

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java [hunk 5]
```diff
@@ -549,22 +545,26 @@ private void mapSPClaimsAndFilterRequestedClaims(Map<String, String> spToLocalCl
...
+                for (Map.Entry<String, String> entry : allLocalClaims.entrySet()) {
+                    //set local2sp role mappings
```
Comment (supunharsha) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java]: review this method

File: components/entitlement/org.wso2.carbon.identity.entitlement.common/src/main/java/org/wso2/carbon/identity/entitlement/common/util/PolicyEditorUtil.java [hunk 6]
```diff
@@ -712,7 +712,7 @@ public static RuleElementDTO createRuleElementDTO(RuleDTO ruleDTO) throws Policy
             throws PolicyEditorException {
 
         List<ObligationElementDTO> obligationElementDTOs = new ArrayList<ObligationElementDTO>();
-        List<Element> returnList = new ArrayList<Element>();
+
```
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement.common/src/main/java/org/wso2/carbon/identity/entitlement/common/util/PolicyEditorUtil.java]: remove the line

File: components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/java/org/wso2/carbon/identity/entitlement/ui/client/EntitlementPolicyUploadExecutor.java [hunk 7]
```diff
@@ -76,15 +76,25 @@ public boolean execute(HttpServletRequest httpServletRequest, HttpServletRespons
...
-                    BufferedReader br = new BufferedReader(new InputStreamReader(fileItem.getDataHandler().getInputStream()));
-                    String temp;
-                    String policyContent = "";
-                    while ((temp = br.readLine()) != null) {
-                        policyContent += temp;
-                    }
-                    if (!"".equals(policyContent)) {
-                        client.uploadPolicy(policyContent);
+                    BufferedReader br = null;
+                    try {
```
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/java/org/wso2/carbon/identity/entitlement/ui/client/EntitlementPolicyUploadExecutor.java]: try-with resourse

File: components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/java/org/wso2/carbon/identity/entitlement/ui/client/EntitlementPolicyUploadExecutor.java [hunk 8]
```diff
@@ -76,15 +76,25 @@ public boolean execute(HttpServletRequest httpServletRequest, HttpServletRespons
...
+                        br = new BufferedReader(new InputStreamReader(fileItem.getDataHandler().getInputStream()));
+                        String temp;
+                        String policyContent = "";
+                        while ((temp = br.readLine()) != null) {
+                            policyContent += temp;
+                        }
+                        if (!"".equals(policyContent)) {
```
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/java/org/wso2/carbon/identity/entitlement/ui/client/EntitlementPolicyUploadExecutor.java]: Use String utils

File: components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/java/org/wso2/carbon/identity/entitlement/ui/client/EntitlementPolicyUploadExecutor.java [hunk 9]
```diff
@@ -76,15 +76,25 @@ public boolean execute(HttpServletRequest httpServletRequest, HttpServletRespons
...
+                            client.uploadPolicy(policyContent);
+                        }
+                    } catch (Exception ex) {
```
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/java/org/wso2/carbon/identity/entitlement/ui/client/EntitlementPolicyUploadExecutor.java]: Check Exception for specific exception

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/EntitlementUtil.java [hunk 10]
```diff
@@ -381,7 +381,7 @@ public static void addSamplePolicies(Registry registry) {
                 + "identity" + File.separator + "policies" + File.separator + "xacml"
                 + File.separator + "default");
 
-        if (policyFolder.exists()) {
+        if (policyFolder.exists() && policyFolder.listFiles() != null) {
```
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/EntitlementUtil.java]: Assign list files to a variable and use

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/cache/DecisionCache.java [hunk 11]
```diff
@@ -87,8 +87,6 @@ public Object getFromCache(String key) {
      */
     public void clearCache() {
         if (log.isDebugEnabled()) {
-            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
-            int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
```
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/cache/DecisionCache.java]: Add tenant domain and id in debug log

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/policy/collection/SimplePolicyCollection.java [hunk 12]
```diff
@@ -162,8 +165,11 @@ public void setPolicyCombiningAlgorithm(PolicyCombiningAlgorithm algorithm) {
 
     @Override
     public boolean deletePolicy(String policyId) {
-
-        return this.policyCollection.remove(policyId) != null ;
+        try {
+            return this.policyCollection.remove(new URI(policyId)) != null;
+        } catch (URISyntaxException ex) {
```
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/policy/collection/SimplePolicyCollection.java]: debug and check

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/handler/AbstractEventHandler.java [hunk 13]
```diff
@@ -53,7 +52,11 @@ public boolean canHandle(MessageContext messageContext) throws IdentityRuntimeEx
             log.error("Error while retrieving event mgt config builder", e);
```
Comment (supunharsha) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/handler/AbstractEventHandler.java]: return false

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/handler/AbstractEventHandler.java [hunk 14]
```diff
@@ -92,7 +95,9 @@ public boolean isAssociationAsync(String eventName) throws IdentityEventExceptio
 
     @Override
     public void init(InitConfig configuration) throws IdentityRuntimeException {
-        this.configs = (ModuleConfiguration)configuration;
+        if (configuration instanceof ModuleConfiguration) {
+            this.configs = (ModuleConfiguration) configuration;
+        }
```
Comment (supunharsha) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/handler/AbstractEventHandler.java]: add exception if casting is not possible

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java [hunk 15]
```diff
@@ -439,32 +441,32 @@ private ClientResponse getAPIResponse(String path, String method, List<Pair> que
             builder = httpClient.resource(url).accept(accept);
         }
 
-        for (String key : headerParams.keySet()) {
-            builder = builder.header(key, headerParams.get(key));
+        for (Entry entry : headerParams.entrySet()) {
```
Comment (supunharsha) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java]: Add type

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java [hunk 16]
```diff
@@ -439,32 +441,32 @@ private ClientResponse getAPIResponse(String path, String method, List<Pair> que
...
+            builder = builder.header(entry.getKey().toString(), entry.getValue());
         }
-        for (String key : defaultHeaderMap.keySet()) {
-            if (!headerParams.containsKey(key)) {
-                builder = builder.header(key, defaultHeaderMap.get(key));
+        for (Entry entry : defaultHeaderMap.entrySet()) {
```
Comment (supunharsha) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java]: Add type

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java [hunk 17]
```diff
@@ -439,32 +441,32 @@ private ClientResponse getAPIResponse(String path, String method, List<Pair> que
...
-        ClientResponse response = null;
+        ClientResponse response;
         String toEncode = IdentityManagementServiceUtil.getInstance().getAppName() + ":" + String.valueOf(IdentityManagementServiceUtil
                 .getInstance().getAppPassword());
         byte[] encoding = Base64.encodeBase64(toEncode.getBytes());
         String authHeader = new String(encoding, Charset.defaultCharset());
 
         if ("GET".equals(method)) {
-            response = (ClientResponse) builder.header("Authorization", "Client "+ authHeader).get(ClientResponse.class);
+            response = (ClientResponse) builder.header("Authorization", "Client " + authHeader).get(ClientResponse.class);
```
Comment (supunharsha) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java]: Add Constants

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java [hunk 18]
```diff
@@ -744,17 +744,15 @@ public boolean doPreUpdateCredential(String userName, Object newCredential,
                 IdentityMgtConfig config = IdentityMgtConfig.getInstance();
                 UserIdentityDataStore identityDataStore = IdentityMgtConfig.getInstance().getIdentityDataStore();
                 UserIdentityClaimsDO identityDTO = identityDataStore.load(userName, userStoreManager);
-                boolean isAccountDisabled = identityDTO.getIsAccountDisabled();
+                boolean isAccountDisabled = false;
+                if (identityDTO != null) {
+                    isAccountDisabled = identityDTO.getIsAccountDisabled();
+                }
 
                 if (isAccountDisabled) {
```
Comment (supunharsha) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java]: merge with if block in line 759

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java [hunk 19]
```diff
@@ -1111,7 +1111,7 @@ public IdentityProvider getEnabledIdPByRealmId(String realmId, String tenantDoma
                 Set<ClaimMapping> returnSet = new HashSet<ClaimMapping>();
                 for (String localClaimURI : localClaimURIs) {
                     for (ClaimMapping claimMapping : claimMappings) {
-                        if (claimMapping.equals(localClaimURI)) {
+                        if (claimMapping.toString().equals(localClaimURI)) {
```
Comment (supunharsha) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java]: Questionable?? if localClaimURI is hash of the object

File: components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/dao/CacheBackedProvisioningMgtDAO.java [hunk 20]
```diff
@@ -54,8 +54,10 @@ public void addProvisioningEntity(String identityProviderName, String connectorT
                                       ProvisioningEntity provisioningEntity, int tenantId, String tenantDomain)
             throws IdentityApplicationManagementException {
 
+
```
Comment (supunharsha) [inline components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/dao/CacheBackedProvisioningMgtDAO.java]: remove empty lines

File: components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/pox/POXSecurityHandler.java [hunk 21]
```diff
@@ -99,11 +99,16 @@ public void init(HandlerDescription description) {
...
         if (msgCtx != null && !msgCtx.isEngaged(POX_SECURITY_MODULE)) {
             return InvocationResponse.CONTINUE;
         }
 
-        AxisService service = msgCtx.getAxisService();
+        AxisService service = null;
+
+        if (msgCtx != null) {
+            service = msgCtx.getAxisService();
+        }
```
Comment (supunharsha) [inline components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/pox/POXSecurityHandler.java]: else throws AxisFault

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java [hunk 22]
```diff
@@ -744,17 +744,15 @@ public boolean doPreUpdateCredential(String userName, Object newCredential,
                 IdentityMgtConfig config = IdentityMgtConfig.getInstance();
                 UserIdentityDataStore identityDataStore = IdentityMgtConfig.getInstance().getIdentityDataStore();
                 UserIdentityClaimsDO identityDTO = identityDataStore.load(userName, userStoreManager);
-                boolean isAccountDisabled = identityDTO.getIsAccountDisabled();
+                boolean isAccountDisabled = false;
+                if (identityDTO != null) {
+                    isAccountDisabled = identityDTO.getIsAccountDisabled();
+                }
```
Comment (supunharsha) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java]: else throw UserStoreException

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java [hunk 23]
```diff
@@ -744,17 +744,15 @@ public boolean doPreUpdateCredential(String userName, Object newCredential,
...
 
                 if (isAccountDisabled) {
                     IdentityErrorMsgContext customErrorMessageContext = new IdentityErrorMsgContext(
                             IdentityCoreConstants.USER_ACCOUNT_DISABLED_ERROR_CODE);
                     IdentityUtil.setIdentityErrorMsg(customErrorMessageContext);
-                } else {
-                    // do nothing
-                }
-                if (identityDTO == null) {
-                    identityDTO = new UserIdentityClaimsDO(userName);
```
Comment (supunharsha) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java]: Check why we assigning new object here

File: components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/listener/IdPMgtValidationListener.java [hunk 24]
```diff
@@ -108,8 +108,8 @@ public boolean doPreUpdateIdP(String oldIdPName, IdentityProvider identityProvid
         }
 
         //Updating a non-shared IdP's name to have shared prefix is not allowed
-        if (oldIdPName != null && !oldIdPName.startsWith(IdPManagementConstants.SHARED_IDP_PREFIX) &&
```
Comment (supunharsha) [inline components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/listener/IdPMgtValidationListener.java]: Check Identitty provider name in line 97

File: components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/OutboundProvisioningManager.java [hunk 25]
```diff
@@ -146,17 +146,19 @@ private static int getTenantIdOfDomain(String tenantDomain)
...
+                // cache hit
+                if (entry != null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Provisioning cache HIT for " + serviceProvider + " of "
+                                + tenantDomainName);
+                    }
+                    return entry.getConnectors();
                 }
-                return entry.getConnectors();
             }
```
Comment (supunharsha) [inline components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/OutboundProvisioningManager.java]: Implement else with exception

File: components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/util/ServicePasswordCallbackHandler.java [hunk 26]
```diff
@@ -305,7 +305,7 @@ private String getPrivateKeyPassword(String username) throws IOException, Except
                         } catch (Exception e) {
```
Comment (supunharsha) [inline components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/util/ServicePasswordCallbackHandler.java]: remove catch block and add debug log

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserRealmProxy.java [hunk 27]
```diff
@@ -1701,33 +1707,25 @@ public void updateRolesOfUser(String userName, String[] roleList) throws UserAdm
...
+                    if (newindex < 0) {
+                        if (realm.getRealmConfiguration().getEveryOneRoleName().equalsIgnoreCase(name)) {
+                            log.warn("Carbon Internal/everyone role can't be manipulated");
+                            continue;
+                        }
+                        delRoles.add(name);
                     }
-                    delRoles.add(name);
                 }
             }
```
Comment (supunharsha) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserRealmProxy.java]: What to do in else case?

File: components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/UserStoreConfigAdminService.java [hunk 28]
```diff
@@ -638,13 +638,17 @@ public boolean accept(File dir, String name) {
                 return name.equalsIgnoreCase(userStoreName);
             }
         });
-        for (File file1 : deleteCandidates) {
-            if (file1.delete()) {
-                log.info("File " + file.getName() + " deleted successfully");
-            } else {
-                log.error("error at deleting file:" + file.getName());
+
+        if (deleteCandidates != null && deleteCandidates.length > 0) {
```
Comment (supunharsha) [inline components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/UserStoreConfigAdminService.java]: Use Array utils

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/servlet/UserAndRoleManagementServlet.java [hunk 29]
```diff
@@ -67,7 +67,7 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
         String category = request.getParameter("category");
         HttpSession session = request.getSession();
 
-        if(category != null && category.equals("users")) {
+        if (category != null && category.equals("users")) {
```
Comment (supunharsha) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/servlet/UserAndRoleManagementServlet.java]: Use string utils

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/servlet/UserAndRoleManagementServlet.java [hunk 30]
```diff
@@ -229,29 +229,27 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
                         List<FlaggedName> dataList = new ArrayList<>(Arrays.asList(datas));
                         exceededDomains = dataList.remove(dataList.size() - 1);
                         session.setAttribute(UserAdminUIConstants.USER_LIST_CACHE_EXCEEDED, exceededDomains);
-                        if (dataList == null || dataList.size() == 0) {
+                        if (dataList.size() == 0) {
```
Comment (supunharsha) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/servlet/UserAndRoleManagementServlet.java]: is empty

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/servlet/UserAndRoleManagementServlet.java [hunk 31]
```diff
@@ -285,14 +283,14 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
                 response.getWriter().write(resp);
 
             }
-        }else if(category!=null && category.equals("roles")){
+        } else if (category != null && category.equals("roles")) {
```
Comment (supunharsha) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/servlet/UserAndRoleManagementServlet.java]: equals and is blank

File: components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/servlet/UserAndRoleManagementServlet.java [hunk 32]
```diff
@@ -319,7 +317,7 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
             session.removeAttribute("previousRole");
             // search filter
             String selectedDomain = request.getParameter("domain");
-            if(selectedDomain == null || selectedDomain.trim().length() == 0){
+            if (selectedDomain == null || selectedDomain.trim().length() == 0) {
```
Comment (supunharsha) [inline components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/servlet/UserAndRoleManagementServlet.java]: Chnage with string utils is blank

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/permission/ManagementPermissionUtil.java [hunk 33]
```diff
@@ -66,25 +67,15 @@ public static void updateRoleUIPermission(String roleName, String[] rawPermissio
 	public static Permission[] getRoleUIPermissions(String roleName, String[] rawPermissions)
 			throws UserAdminException {
 		Permission[] permissions = new Permission[0];
-		try {
-
-			if (rawPermissions == null || rawPermissions.length == 0) {
-				return permissions;
-			}
+		if (ArrayUtils.isEmpty(permissions)) {
```
Comment (rushmin) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/permission/ManagementPermissionUtil.java]: Return an empty array for the sake of readability.

File: components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt.ui/src/main/java/org/wso2/carbon/identity/workflow/mgt/ui/util/WorkflowUIUtil.java [hunk 34]
```diff
@@ -78,20 +77,19 @@ public static  void loadTemplateParameters(Map<String, String[]> requestParamete
      * @param requestParameterMap parameter map
      * @param workflowWizard      Workflow object
      */
-    public static  void loadWorkflowImplParameters(Map<String, String[]> requestParameterMap, WorkflowWizard workflowWizard){
-        Set<String> keys = requestParameterMap.keySet();
+    public static  void loadWorkflowImplParameters(Map<String, String[]> requestParameterMap, WorkflowWizard workflowWizard) {
         Parameter[] workflowImplParameters = workflowWizard.getWorkflowImplParameters();
```
Comment (rushmin) [inline components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt.ui/src/main/java/org/wso2/carbon/identity/workflow/mgt/ui/util/WorkflowUIUtil.java]: Add a new line between the method and the implementation.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java (hunk 2 above)
Comment (supunharsha) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java]: fixed

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java (hunk 4 above)
Comment (supunharsha) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java]: fixed

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java (hunk 5 above)
Comment (supunharsha) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java]: Fixed. Need to verify this

File: components/entitlement/org.wso2.carbon.identity.entitlement.common/src/main/java/org/wso2/carbon/identity/entitlement/common/util/PolicyEditorUtil.java (hunk 6 above)
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement.common/src/main/java/org/wso2/carbon/identity/entitlement/common/util/PolicyEditorUtil.java]: fixed

File: components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/java/org/wso2/carbon/identity/entitlement/ui/client/EntitlementPolicyUploadExecutor.java (hunk 7 above)
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/java/org/wso2/carbon/identity/entitlement/ui/client/EntitlementPolicyUploadExecutor.java]: fixed

File: components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/java/org/wso2/carbon/identity/entitlement/ui/client/EntitlementPolicyUploadExecutor.java (hunk 8 above)
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/java/org/wso2/carbon/identity/entitlement/ui/client/EntitlementPolicyUploadExecutor.java]: ? isNotBlank or IsNotEmpty

File: components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/java/org/wso2/carbon/identity/entitlement/ui/client/EntitlementPolicyUploadExecutor.java (hunk 9 above)
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement.ui/src/main/java/org/wso2/carbon/identity/entitlement/ui/client/EntitlementPolicyUploadExecutor.java]: Fixed with IOException

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/EntitlementUtil.java (hunk 10 above)
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/EntitlementUtil.java]: fixed

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/cache/DecisionCache.java (hunk 11 above)
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/cache/DecisionCache.java]: fixed

File: components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/internal/EntitlementServiceComponent.java [hunk 35]
```diff
@@ -235,7 +235,7 @@ protected void activate(ComponentContext ctxt) {
 
                 boolean customPolicies = false;
 
-                if (policyFolder != null && policyFolder.exists()) {
+                if (policyFolder != null && policyFolder.exists() && policyFolder.listFiles() != null) {
```
Comment (supunharsha) [inline components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/internal/EntitlementServiceComponent.java]: fixed

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/handler/AbstractEventHandler.java (hunk 13 above)
Comment (supunharsha) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/handler/AbstractEventHandler.java]: fixed

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/handler/AbstractEventHandler.java (hunk 14 above)
Comment (supunharsha) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/handler/AbstractEventHandler.java]: fixed

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java [hunk 36]
```diff
@@ -258,9 +258,11 @@ public String parameterToString(Object param) {
         List<Pair> params = new ArrayList<Pair>();
 
         // preconditions
-        if (name == null || name.isEmpty() || value == null) return params;
+        if (name == null || name.isEmpty() || value == null) {
```
Comment (supunharsha) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java]: used string utils

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java (hunk 15 above)
Comment (supunharsha) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java]: fixed

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java (hunk 16 above)
Comment (supunharsha) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java]: fixed

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java (hunk 17 above)
Comment (supunharsha) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java]: fixed

**Pull request merged**